test/hotspot/jtreg/serviceability/sa/UniqueVtableTest.java

```diff
@@ -131,26 +144,31 @@ private void runTest() throws Throwable {
         }
     }
 
-    private void run() throws Throwable {
+    private static void createAnotherToAttach(long lingeredAppPid) throws Throwable {
+        // Start a new process to attach to the lingered app
+        ProcessBuilder processBuilder = ProcessTools.createJavaProcessBuilder(
+            "--add-modules=jdk.hotspot.agent",
+            "--add-exports=jdk.hotspot.agent/sun.jvm.hotspot=ALL-UNNAMED",
+            "--add-exports=jdk.hotspot.agent/sun.jvm.hotspot.debugger=ALL-UNNAMED",
+            "--add-exports=jdk.hotspot.agent/sun.jvm.hotspot.types=ALL-UNNAMED",
+            "--add-exports=jdk.hotspot.agent/sun.jvm.hotspot.types.basic=ALL-UNNAMED",
+            "UniqueVtableTest",
+            Long.toString(lingeredAppPid));
+        SATestUtils.addPrivilegesIfNeeded(processBuilder);
+        OutputAnalyzer SAOutput = ProcessTools.executeProcess(processBuilder);
+        SAOutput.shouldHaveExitValue(0);
+        System.out.println(SAOutput.getOutput());
+    }
+
+    private static void runMain() throws Throwable {
         Throwable reasonToFail = null;
         LingeredApp app = null;
         try {
             app = LingeredApp.startApp();
-            attach(app.getPid());
-            runTest();
+            createAnotherToAttach(app.getPid());
```

Comment:
Did you ever figure out why attaching from the main test process sometimes fails?

Reply:
No. It looks like JTReg executes main test process differently than the sub-process is executed, but I didn't find difference which can cause attach timeouts.